### ISSUE TITLE
CrossOver doesn't work in Fullscreen mode

### ISSUE THREAD

**Issue by alecs91** [lacymorrow/crossover#64]
It seems that CrossOver doesn't work on EFT:
As soon as i set up the crosshair and then alt-tab in the EFT screen the crosshair disappears. 
If i press any keyboard shortcut like alt+ctrl+maiusc+x (or 'h', or whatever) the game screen minimize and it brings me back to the desktop.
Suggestions?
Edit: It works with "screen mode borderless". It doesn't work with "screen mode fullscreen"

**Comment by lacymorrow** [comment]
@alecs91 Unfortunately fullscreen mode doesn't work in most games, however there is a workaround using AutoHotKey https://github.com/lacymorrow/crossover#workaround

**Comment by kyle-go** [comment]
@lacymorrow I found a way to work in fullscreen mode.
You can find the code in the link blow.

[https://github.com/electron/electron/issues/26350#issuecomment-1128813997](https://github.com/electron/electron/issues/26350#issuecomment-1128813997)

**Comment by lacymorrow** [comment]
@kyle-go What OS and Electron version are you using?

Your example code is almost identical to what I use with crossover already, unfortunately.

I would LOVE to find a workaround to this issue though.

**Comment by AndrazPustotnik** [comment]
Any news here yet? :/

**Comment by lacymorrow** [comment]
This is the issue blocking things: https://github.com/electron/electron/issues/26350#issuecomment-1128813997

**Comment by AndrazPustotnik** [comment]
I digged into it a little bit. And since fullscreen mode gives app the ability to use the graphic card in fully it means that in order to do overlay on top of that, some kind of gpu library would need to be used like DirectX or OpenGL. It makes whole problem a lot more complicated and if we would be using it on top of video game we may have problems with anti cheat stuff... :/  
Ask Discord how they did it :P

**Comment by lacymorrow** [comment]
Most likely a proprietary native extension, which would be hard for me to support as the sole dev on the project.

**Comment by GreenMan36** [comment]
I noticed my Steam popped up with a friend launching [Crosshair X](https://store.steampowered.com/app/1366800/Crosshair_X/) and I went to check it out and tell him CrossOver is free.
But then I noticed [this update post](https://store.steampowered.com/app/1366800/Crosshair_X/) which says they have Exclusive Fullscreen support thanks to the Xbox Game Bar Extension support on Windows 10 and 11.

Maybe this can be looked into? They still have it in Beta but clearly this is an option.
It may require significant work but I think its worth researching, maybe even making an issue or project board for it?

**Comment by lacymorrow** [comment]
Definitely worth researching! Thanks for flagging!

**Comment by k9dog-gmail** [comment]
Found a workaround X (at least xfce4) windows have an option to go full screen .. now you can alt-tab to browser etc

**Comment by websharik** [comment]
FPS Monitor can working with fullscreen apps. How ?

**Comment by lacymorrow** [comment]
@websharik It's written in another language and is injecting itself directly into Vulkan/DirectX.

It's possible (discord does it), but very difficult for one person to maintain in their free time.

**Comment by lacymorrow** [comment]
@GreenMan36 Can I ask what your friend likes about Crosshair X?
I'd love to catch up to the competition... for free

**Comment by GreenMan36** [comment]
> @GreenMan36 Can I ask what your friend likes about Crosshair X? I'd love to catch up to the competition... for free

I have no clue, I can't remember which friend was using it. A simple plus is that it's available on Steam, with auto updates. Community workshop. The UI looks decent and you can share your configs. It seems to have great crosshair creator. Lots of settings. 

One thing I'm not seeing is game-specific crosshair settings. That would be great to have. Auto-hiding and changing the crosshair based on what program is focussed could be super-b.

**Comment by Pen9G** [comment]
Hello, i think Xbox Game Bar Widget is a good solution for Fullscreen game crosshair overlay. I have used some crosshair widgets on Xbox Game Bar myself, and i tried develop a Xbox Game Bar Crosshair Overlay Widget myself last week, its quite simple because Xbox Game Bar Widget is a common UWP application.
 
The only problem may be that Xbox Game Bar Widget cannot listen to background mouse events, so "Hide on ADS" needs an extra desktop app to communicate with the Xbox Game Bar Widget. As i know the Xbox Game Bar Version of Crosshair X need the desktop app to run at same time too.

**Comment by Danalytics-dan** [comment]
"> Hello, i think Xbox Game Bar Widget is a good solution for Fullscreen game crosshair overlay. I have used some crosshair widgets on Xbox Game Bar myself, and i tried develop a Xbox Game Bar Crosshair Overlay Widget myself last week, its quite simple because Xbox Game Bar Widget is a common UWP application.
> 
> The only problem may be that Xbox Game Bar Widget cannot listen to background mouse events, so "Hide on ADS" needs an extra desktop app to communicate with the Xbox Game Bar Widget. As i know the Xbox Game Bar Version of Crosshair X need the desktop app to run at same time too."


It would be a good solution, however there is no target audience while using it.
Xbox game bar uses performance since it has to run while the game is running and so anyone trying to squeeze all performance out of their system will disable it, myself included.

If you don't care about performance and have game bar enabled then the      pre-existing electron method is just as good, since you only have to go to bordered fullscreen (which reduces performance, about as much as game bar)

tldr:
-Isn't good for those who want maximum performance
-Is equally good as the existing technique for everyone else
So no-one benefits.

**Comment by Pen9G** [comment]
> tldr:
>-Isn't good for those who want maximum performance
>-Is equally good as the existing technique for everyone else
>So no-one benefits.

I actually don't think Xbox Game Bar will cause any obvious game performance reduce, at least compared to toggling Exclusive Fullscreen to Fullscreen. I've been using Xbox Game Bar in games like Hunt: Showdown, and I never notice any FPS drop after activating Xbox Game Bar(I make sure no Game Bar process was running).

Besides, the fact that some Crosshair overlay app such as CrosshairX have their own Game Bar edition indicates the advantages of Xbox Game Bar over solutions just not using Exclusive Fullscreen, right?

**Comment by Danalytics-dan** [comment]
> I actually don't think Xbox Game Bar will cause any obvious game performance reduce, at least compared to toggling Exclusive Fullscreen to Fullscreen. I've been using Xbox Game Bar in games like Hunt: Showdown, and I never notice any FPS drop after activating Xbox Game Bar(I make sure no Game Bar process was running).
> 
> Besides, the fact that some Crosshair overlay app such as CrosshairX have their own Game Bar edition indicates the advantages of Xbox Game Bar over solutions just not using Exclusive Fullscreen, right?

Xbox game bar cannot be disabled apart from with special tools, such as Atlas OS or with tweaking software. People trying to squeeze every % of performance out of their pc will try to reduce as many windows background processes as possible including game bar.

While game bar might not lose quite as much fps as Exclusive Fullscreen it still will do and so for those trying to get as much performance it isn't viable.

tldr:
-slightly better for those who don't care about performance
-people who care seriously about performance still won't use it
-takes effort to implement

Having said that, a toggle-able option would be better than nothing of course for the niche people it would benefit.

**Comment by lacymorrow** [comment]
📋 Reviewed by Paperclip — categorized as discussion/community thread/wontfix. No task created.

**Comment by lacymorrow** [comment]
📋 Tracked in Paperclip as **LAC-1810**. This issue has been added to our internal project management system and will be reviewed by the engineering team.

**Comment by lacymorrow** [comment]
Tracked in Paperclip as **LAC-2590**. This issue has been added to our internal project management system and will be reviewed by the engineering team.

**Comment by lacymorrow** [comment]
Exclusive fullscreen modes can block overlays — that's a fundamental OS/GPU constraint, not a CrossOver bug. Use borderless windowed mode for the best overlay experience. This is documented in the README.
